ca/ca.go

```diff
@@ -427,13 +432,60 @@ func (ca *CertificateAuthorityImpl) IssuePrecertificate(ctx context.Context, iss
 		return nil, err
 	}
 
-	precertDER, err := ca.issuePrecertificateInner(ctx, issueReq, serialBigInt, validity, precertType)
-	if err != nil {
-		return nil, err
+	if features.Enabled(features.PrecertificateOCSP) {
+		serialHex := core.SerialToString(serialBigInt)
+		nowNanos := ca.clk.Now().UnixNano()
+		expiresNanos := validity.NotAfter.UnixNano()
+		_, err = ca.sa.AddSerial(ctx, &sapb.AddSerialRequest{
+			Serial:  &serialHex,
+			RegID:   issueReq.RegistrationID,
+			Created: &nowNanos,
+			Expires: &expiresNanos,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		precertDER, err := ca.issuePrecertificateInner(ctx, issueReq, serialBigInt, validity, precertType)
+		if err != nil {
+			return nil, err
+		}
+
+		ocspResp, err := ca.GenerateOCSP(ctx, core.OCSPSigningRequest{
+			CertDER: precertDER,
+			Status:  "good",
+		})
+		if err != nil {
+			err = berrors.InternalServerError(err.Error())
+			ca.log.AuditInfof("OCSP Signing failure: serial=[%s] err=[%s]", core.SerialToString(serialBigInt), err)
+		}
+
+		_, err = ca.sa.AddPrecertificate(ctx, &sapb.AddCertificateRequest{
+			Der:    precertDER,
+			RegID:  issueReq.RegistrationID,
+			Ocsp:   ocspResp,
+			Issued: &nowNanos,
+		})
+		if err != nil {
```

Comment:
It seems like this is a case where we'd want to queue an `orphanedCert{}` - is there a reason it can't support pre-certificates right now? Should there be a follow-up issue to extend it for precerts?

Reply:
Filed a followup issue: https://github.com/letsencrypt/boulder/issues/4425. The orphan queue isn't quite right for this as it stands because it would wind up inserting the precertificate into the final certificates table.